### PR TITLE
jenkins: add postject pipeline

### PR DESCRIPTION
**Pull request by RaisinTen**
Fixes: https://github.com/nodejs/postject/issues/62
Signed-off-by: Darshan Sen <raisinten@gmail.com>

cc @nodejs/single-executable

### CODE REVIEW

File: jenkins/pipelines/postject.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+
+        stage('Build postject') {
+            steps {
+                // Calling with `returnStatus` suppresses automatic failures
+                sh(script: "npm install", returnStatus: true)
+                sh(script: "git clone https://github.com/emscripten-core/emsdk.git", returnStatus: true)
+                sh(script: "(cd emsdk && ./emsdk install latest)", returnStatus: true)
+                sh(script: "(cd emsdk && ./emsdk activate latest)", returnStatus: true)
+                sh(script: "(cd emsdk && source ./emsdk_env.sh && cd .. && npm run build -- --jobs=3)", returnStatus: true)
```
Comment (ovflowd) [inline jenkins/pipelines/postject.jenkinsfile:46]: What if this step fails? What `returnStatus` supposedly does?ß

**Review by ovflowd — APPROVED** [review]
Code-wise it looks nice!

**Review by aymen94 — APPROVED** [review]

File: jenkins/pipelines/postject.jenkinsfile (hunk 1 above)
Comment (tony-go) [reply jenkins/pipelines/postject.jenkinsfile:46]: > Normally, a script which exits with a nonzero status code will cause the step to fail with an exception. If this option is checked, the return value of the step will instead be the status code. You may then compare it to zero, for example. 

From: https://www.jenkins.io/doc/pipeline/steps/workflow-durable-task-step/

As we didn't check it in any, we could probably remove it. WDYT @RaisinTen ?

File: jenkins/pipelines/postject.jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        }
+
+
+        stage('Build postject') {
+            steps {
+                // Calling with `returnStatus` suppresses automatic failures
+                sh(script: "npm install", returnStatus: true)
+                sh(script: "git clone https://github.com/emscripten-core/emsdk.git", returnStatus: true)
+                sh(script: "(cd emsdk && ./emsdk install latest)", returnStatus: true)
+                sh(script: "(cd emsdk && ./emsdk activate latest)", returnStatus: true)
```
Comment (targos) [inline jenkins/pipelines/postject.jenkinsfile:45]: Does this take a small enough amount of time that we can do it every time?

**Comment by RaisinTen** [comment]
I've heard from @mhdawson that we don't use pipelines a lot, so we worked on creating this job instead - https://ci.nodejs.org/view/All/job/node-test-postject. See https://github.com/nodejs/build/issues/3126 for more info on that. Closing this in favor of that job.

**Pull request closed without merging**